chalk-ir/src/lib.rs

```diff
@@ -811,6 +811,12 @@ pub struct AliasEq<I: Interner> {
     pub ty: Ty<I>,
 }
 
+impl<I: Interner> AliasEq<I> {
+    pub fn interner(&self) -> &I {
```

Comment:
Yeah, I don't think this function makes sense here.

Reply:
It has to either come from the AliasEq or it has to be part of the trait and no other implementor of this trait uses the interner yet.

Reply:
@detrumi is working the `impl Trait`/`Alias`, so this portion of the API might change. But regardless, even if the other types don't use it and we still need it, I don't think having `AliasEq` store the interner is correct.